engine/src/main/java/org/kigalisim/engine/support/PopulationChangeRecalcStrategy.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Strategy for recalculating population changes.
+ *
+ * <p>This strategy encapsulates the logic previously found in the
+ * recalcPopulationChange method of SingleThreadEngine.</p>
+ *
+ * @license BSD-3-Clause
+ */
+
+package org.kigalisim.engine.support;
+
+import java.math.BigDecimal;
+import org.kigalisim.engine.Engine;
+import org.kigalisim.engine.number.EngineNumber;
+import org.kigalisim.engine.number.UnitConverter;
+import org.kigalisim.engine.state.OverridingConverterStateGetter;
+import org.kigalisim.engine.state.Scope;
+
+/**
+ * Strategy for recalculating population changes.
+ */
+public class PopulationChangeRecalcStrategy implements RecalcStrategy {
+
+  private Scope scope;
+  private Boolean subtractRecharge;
+
+  /**
+   * Create a new PopulationChangeRecalcStrategy.
+   *
+   * @param scope The scope to use for calculations, null to use engine's current scope
+   * @param subtractRecharge Whether to subtract recharge, null to default to true
+   */
+  public PopulationChangeRecalcStrategy(Scope scope, Boolean subtractRecharge) {
+    this.scope = scope;
+    this.subtractRecharge = subtractRecharge;
+  }
+
+  @Override
+  public void execute(Engine target, RecalcKit kit) {
+    // Move the logic from SingleThreadEngine.recalcPopulationChange
+    OverridingConverterStateGetter stateGetter =
+        new OverridingConverterStateGetter(kit.getStateGetter());
```

Comment:
Please save kit.getStateGetter to a local variable and then use here in this line to simplify. If possible, please make this one line instead of using a line continuation.